News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/NewsReaderListActivity.java

```diff
@@ -1133,35 +1133,79 @@ private void UpdateListView() {
 
 	@Override
 	public void onClick(RssItemViewHolder vh, int position) {
+		Feed feed = vh.getRssItem().getFeed();
+		Long openIn	= feed.getOpenIn();
+
+		Uri currentUrl = Uri.parse(vh.getRssItem().getLink());
+
+		if (openIn == null) {
+			if (mPrefs.getBoolean(SettingsActivity.CB_SKIP_DETAILVIEW_AND_OPEN_BROWSER_DIRECTLY_STRING, false)) {
+
+				//Choose Browser based on user settings
+				//modified copy from NewsDetailFragment.java:loadUrl(String url)
+				int selectedBrowser = Integer.parseInt(mPrefs.getString(SettingsActivity.SP_DISPLAY_BROWSER, "0"));
+				switch(selectedBrowser) {
+					case 0:
+						openRssItemInCustomTab(currentUrl);
+						break;
+					case 1:
+						//openRssItemInInternalBrowser(currentUrl);
+						break;
+					case 2:
+						openRssItemInExternalBrowser(currentUrl);
+						break;
+				}
 
-		if (mPrefs.getBoolean(SettingsActivity.CB_SKIP_DETAILVIEW_AND_OPEN_BROWSER_DIRECTLY_STRING, false)) {
-			String currentUrl = vh.getRssItem().getLink();
-
-			//Choose Browser based on user settings
-			//modified copy from NewsDetailFragment.java:loadUrl(String url)
-			int selectedBrowser = Integer.parseInt(mPrefs.getString(SettingsActivity.SP_DISPLAY_BROWSER, "0"));
-			if (selectedBrowser == 0) { // Custom Tabs
-				CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder()
-						.setShowTitle(true)
-						.setStartAnimations(this, R.anim.slide_in_right, R.anim.slide_out_left)
-						.setExitAnimations(this, R.anim.slide_in_left, R.anim.slide_out_right)
-						.addDefaultShareMenuItem();
-				builder.build().launchUrl(this, Uri.parse(currentUrl));
-			} else { //External browser
-				Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(currentUrl));
-				startActivity(browserIntent);
+				((NewsListRecyclerAdapter) getNewsReaderDetailFragment().getRecyclerView().getAdapter()).changeReadStateOfItem(vh, true);
+			} else {
+				openRssItemInDetailedView(position);
 			}
 
-			((NewsListRecyclerAdapter) getNewsReaderDetailFragment().getRecyclerView().getAdapter()).changeReadStateOfItem(vh, true);
-		} else {
-			Intent intentNewsDetailAct = new Intent(this, NewsDetailActivity.class);
+			return;
+		}
 
-			intentNewsDetailAct.putExtra(NewsReaderListActivity.ITEM_ID, position);
-			intentNewsDetailAct.putExtra(NewsReaderListActivity.TITLE, getNewsReaderDetailFragment().getTitle());
-			startActivityForResult(intentNewsDetailAct, Activity.RESULT_CANCELED);
+		switch (openIn.intValue()) {
+			case 1:
+				openRssItemInDetailedView(position);
+				break;
+			case 2:
+				openRssItemInCustomTab(currentUrl);
+				break;
+			case 3:
+				openRssItemInExternalBrowser(currentUrl);
+				break;
+			default:
+				Log.e("NRLA", "openIn has illegal value");
```

Comment:
maybe we can also throw an error here? just like you did [here](https://github.com/nextcloud/news-android/pull/1266/files#diff-c55715a6b6445200e276f884d9201180715d6bbac9b155526b1ba29902d12ff3R313)
```suggestion
				throw new RuntimeException("Unreachable: openIn has illegal value " + openIn);
```